### PR TITLE
Cortx 28183 part2

### PR DESCRIPTION
**Pull request by pragamj**
# Problem Statement
- Added param for ext_lb_config and check for dir exists in pre-req

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [x] Attach test execution logs
    https://eos-jenkins.colo.seagate.com/job/QA/job/K8s_Cortx_Continuous_Deployment/690/console
https://eos-jenkins.colo.seagate.com/job/QA/job/K8s_Cortx_Continuous_Deployment_sns_config/590/console
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)
[collection.log](https://github.com/Seagate/cortx-test/files/8359524/collection.log)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by priyankaborawake — APPROVED** [review]

**Review by swapnil-khandare — APPROVED** [review]

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 1]
```diff
@@ -222,14 +224,17 @@ def execute_prereq_cortx(self, node_obj: LogicalNode, remote_code_path: str, sys
...
-        resp1 = node_obj.execute_cmd(cmd="ls -lhR /mnt/fs-local-volume/", read_lines=True)
-        LOGGER.info("\n %s", resp1)
-        openldap_dir_residue = node_obj.execute_cmd(cmd="ls -lhR /etc/3rd-party/",
-                                                    read_lines=True)
-        LOGGER.info("\n %s", openldap_dir_residue)
-        thirdparty_residue = node_obj.execute_cmd(cmd="ls -lhR /var/data/3rd-party/",
-                                                  read_lines=True)
-        LOGGER.info("\n %s", thirdparty_residue)
+        if node_obj.path_exists("/mnt/fs-local-volume/"):
+            resp1 = node_obj.execute_cmd(cmd="ls -lhR /mnt/fs-local-volume/", read_lines=True)
```
Comment (sonalk0209) [inline libs/prov/prov_k8s_cortx_deploy.py:228]: can commands go to commands.py?

File: config/prov/prov_test.yaml [hunk 2]
```diff
@@ -128,11 +128,14 @@ k8s_cortx_deploy:
...
   setup_size: "small"
   protocol: "http://{}:{}"
+  https_protocol: "https://{}"
   https_port: 30443
   http_port: 30080
   control_port_https: 31169
   iface: "eth1"
   service_type: "LoadBalancer"
+  lb_count: 1
+  deployment_type: "standard"
```
Comment (sonalk0209) [inline config/prov/prov_test.yaml:138]: just for my understanding, what other options for deployment type?

File: config/prov/prov_test.yaml [hunk 3]
```diff
@@ -128,11 +128,14 @@ k8s_cortx_deploy:
   s3_max_start_timeout: 240
   setup_size: "small"
   protocol: "http://{}:{}"
+  https_protocol: "https://{}"
```
[the remark this answers is not in the document]
Comment (pragamj) [reply config/prov/prov_test.yaml:131]: its with lb

File: config/prov/prov_test.yaml (hunk 2 above)
Comment (pragamj) [reply config/prov/prov_test.yaml:138]: we have 2 options standard and data-only

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 4]
```diff
@@ -1210,17 +1226,16 @@ def test_deployment(self, sns_data, sns_parity,
                 if not resp[0]:
                     LOGGER.debug("Did not get expected response: %s", resp)
                 ext_ip = resp[1]
-                port = resp[3]
+                port = resp[2]
```
Comment (sonalk0209) [inline libs/prov/prov_k8s_cortx_deploy.py:1229]: with resp[2] we would need to chnage the endpoint as well to https://<ip>:<port>, is that taken care?

File: config/prov/prov_test.yaml [hunk 5]
```diff
@@ -123,16 +123,21 @@ k8s_cortx_deploy:
...
   https_port: 30443
   http_port: 30080
   control_port_https: 31169
   iface: "eth1"
   service_type: "LoadBalancer"
+  lb_count: 1
```
Comment (sonalk0209) [inline config/prov/prov_test.yaml:139]: so in case of loadbalancer this count will come from??

Comment (pragamj) [reply config/prov/prov_test.yaml:139]: Jenkins job

**Review by sonalk0209 — APPROVED** [review]

**Review by seagate-sarang-sawant — APPROVED** [review]

**Pull request merged**